### PR TITLE
Add AWS S3 storage

### PR DESCRIPTION
**Pull request by jokly**
PR for the issue #3 
[Telegram Profile](https://t.me/jokly32)

### CODE REVIEW

File: package-lock.json
Comment (krigga) [inline package-lock.json:1]: This project uses yarn, so please remove this

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "dist/**/*"
   ],
   "dependencies": {
+    "@aws-sdk/client-s3": "^3.421.0",
```
Comment (krigga) [inline package.json:13]: There is a new version, please update

File: src/storage.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import PinataClient from '@pinata/sdk';
 import { Readable } from 'stream';
+import { S3 } from "@aws-sdk/client-s3"
```
Comment (krigga) [inline src/storage.ts:3]: Please use single quotes and add a semicolon

File: src/storage.ts [hunk 3]
```diff
@@ -1,5 +1,7 @@
...
+import { AwsCredentialIdentity } from "@aws-sdk/types";
```
Comment (krigga) [inline src/storage.ts:4]: Since this type is not needed (see below), please remove this

File: src/storage.ts [hunk 4]
```diff
@@ -20,3 +22,30 @@ export class PinataStorage implements Storage {
...
 }
+
+export class S3Storage implements Storage {
+    private readonly bucket: string;
+    private readonly s3: S3;
+
+    constructor(accessKeyId: string, secretAccessKey: string, bucket: string) {
+        this.bucket = bucket;
+        this.s3 = new S3({
+            credentials: {
```
Comment (krigga) [inline src/storage.ts:33]: This object can be written as just
```ts
{
                accessKeyId,
                secretAccessKey,
}
```
(no need to put `!` and do type assertions)

File: src/storage.ts [hunk 5]
```diff
@@ -20,3 +22,30 @@ export class PinataStorage implements Storage {
...
+                accessKeyId: accessKeyId!,
+                secretAccessKey: secretAccessKey!,
+            } as AwsCredentialIdentity,
+        });
+    }
+
+    async uploadFile(contents: Buffer): Promise<string> {
```
Comment (krigga) [inline src/storage.ts:40]: Please add 2 missing semicolons in this method

**Review by krigga — CHANGES_REQUESTED** [review]

**Comment by jokly** [comment]
Thank you for the review. Fixed the problems

**Review by krigga — CHANGES_REQUESTED** [review]
I'm sorry, I forgot to tell you the first time, this storage option also needs to be added here https://github.com/ton-community/gamefi-sdk/blob/e56f2bcedfc08aca52c8d2e3deb92d5704541133/src/sdk.ts#L18-L30

**Comment by jokly** [comment]
Done :)

**Review by krigga — APPROVED** [review]
LGTM, thank you for your contribution!

**Pull request merged**